index.js

```diff
@@ -1,27 +1,154 @@
+//   });
+
+//Iterarion 3: insert ALLrecipes
+
+// mongoose
+//   .connect(MONGODB_URI, {
+//     useCreateIndex: true,
+//     useNewUrlParser: true,
+//     useUnifiedTopology: true
+//   })
+//   .then(self => {
+//     console.log(`Connected to the database: "${self.connection.name}"`);
+//     // Before adding any documents to the database, let's delete all previous entries
+//     return self.connection.dropDatabase();
+//   })
+//   .then(() => {
+//     Recipe.insertMany(data)
+//     .then([...data].forEach(elem => {console.log("Inserted: ", elem.title)}))
+//     .catch(error => {
+//       console.error('Error on insertion', error)});
+//   })
+//   .then( mongoose.connection.close(() => console.log(`Connection closed`)))
+//   .catch(error => {
+//     console.error('Error connecting to the database', error);
+//   });
+
+
+// //Iteration 4:
+
+// mongoose
+//   .connect(MONGODB_URI, {
+//     useCreateIndex: true,
+//     useNewUrlParser: true,
+//     useUnifiedTopology: true
+//   })
+//   .then(self => {
+//     console.log(`Connected to the database: "${self.connection.name}"`);
+//   })
+//   .then(() => {
+//     Recipe.findOneAndUpdate({title: "Rigatoni alla Genovese"}, {duration: 100}, (error, recipe) => {
+//       if (error) {
+//         console.log("Error on update: ", error);
+//       } else {
+//         console.log("Recipe successfully updated: ", recipe);
+//       }
+//     });
+//   })
+//   .then( mongoose.connection.close(() => console.log(`Connection closed`)))
+//   .catch(error => {
+//     console.error('Error connecting to the database', error);
+//   });
+
+// //Iteration 5:
+
+// mongoose
+//   .connect(MONGODB_URI, {
+//     useCreateIndex: true,
+//     useNewUrlParser: true,
+//     useUnifiedTopology: true
+//   })
+//   .then(self => {
+//     console.log(`Connected to the database: "${self.connection.name}"`);
+//   })
+//   .then(() => {
+//     Recipe.deleteOne({title: "Carrot Cake"}, (error) => {
+//       if (error) {
+//         console.log("Error on deletion: ", error);
+//       } else {
+//         console.log("Recipe successfully deleted");
+//       }
+//     });
+//   })
+//   .then( mongoose.connection.close(() => console.log(`Connection closed`)))
+//   .catch(error => {
+//     console.error('Error connecting to the database', error);
+//   });
+
+  //Iteration 6:
+
+  mongoose
   .connect(MONGODB_URI, {
     useCreateIndex: true,
     useNewUrlParser: true,
     useUnifiedTopology: true
   })
   .then(self => {
     console.log(`Connected to the database: "${self.connection.name}"`);
-    // Before adding any documents to the database, let's delete all previous entries
-    return self.connection.dropDatabase();
-  })
-  .then(() => {
-    // Run your code here, after you have insured that the connection was made
-  })
-  .catch(error => {
+    return self.connection.dropDatabase()
+    .then(() => {
+      Recipe.insertMany(data)
+      .then( () => {
```

Comment:
That's fancy!!! I suppouse you have your own recipe in the data.json, right?! Nice job dude! You made it better than me...